### PR TITLE
Add a torch function similar to numpy.in1d

### LINKED ISSUE #3025: Function request: np.isin
I believe that should be useful to have a function similar to https://docs.scipy.org/doc/numpy-1.13.0/reference/generated/numpy.in1d.html, that compares a tensor element-wise with a list of possible values. (I'm using it to filter labels/classes in some classifiers).

Expected behavior:

```
>>> a = torch.LongTensor([[1,2,3],[1,1,2],[3,5,1]])
>>> a
 1  2  3
 1  1  2
 3  5  1
[torch.LongTensor of size 3x3]
>>> a.in(torch.LongTensor([1, 2, 5]))
 1 1 0
 1 1 1
 0 1 1
```

Now it's possible to implement it by iterating over the filter, and storing the results in a tensor with the OR operator. But a faster implementation is possible with TH/THC

My current implementation:

```
@utils.tensorfy(0, 1, tensor_klass=torch.LongTensor)
def filter_labels(y, labels):
    """Utility used to create a mask to filter values in a tensor.

    Args:
        y (list, torch.Tensor): tensor where each element is a numeric integer
            representing a label.
        labels (list, torch.Tensor): filter used to generate the mask. For each
            value in ``y`` its mask will be "1" if its value is in ``labels``,
            "0" otherwise".

    Shape:
        y: can have any shape. Usually will be :math:`(N, S)` or :math:`(S)`,
            containing `batch X samples` or just a list of `samples`.
        labels: a flatten list, or a 1D LongTensor.

    Returns:
        mask (torch.ByteTensor): a binary mask, with "1" with the respective value from ``y`` is
        in the ``labels`` filter.

    Example::

        >>> a = torch.LongTensor([[1,2,3],[1,1,2],[3,5,1]])
        >>> a
         1  2  3
         1  1  2
         3  5  1
        [torch.LongTensor of size 3x3]
        >>> classification.filter_labels(a, [1, 2, 5])
         1  1  0
         1  1  1
         0  1  1
        [torch.ByteTensor of size 3x3]
        >>> classification.filter_labels(a, torch.LongTensor([1]))
         1  0  0
         1  1  0
         0  0  1
        [torch.ByteTensor of size 3x3]
    """
    mapping = torch.zeros(y.size()).byte()

    for label in labels:
        mapping = mapping | y.eq(label)

    return mapping

```

cc @mruberry @rgommers @heitorschueroff

### PR DESCRIPTION
**Pull request by CamiWilliams**
PR to have a function similar to https://docs.scipy.org/doc/numpy-1.13.0/reference/generated/numpy.in1d.html, that compares a tensor element-wise with a list of possible values.

```
>>> a = torch.LongTensor([[1,2,3],[1,1,2],[3,5,1]])
>>> a
 1  2  3
 1  1  2
 3  5  1
[torch.LongTensor of size 3x3]
>>> a.in(torch.LongTensor([1, 2, 5]))
 1 1 0
 1 1 1
 0 1 1
[torch.ByteTensor of size 3x3]
```

resolve #3025

### CODE REVIEW

**Review by ezyang — CHANGES_REQUESTED** [review]
Hi @CamiWilliams! I think the request in the original issue was to actually implement a kernel that is faster than the pure Python implementation here. There is a small amount of utility to adding a slow implementation of a missing API, but in general it is better if things PyTorch library are at least reasonably fast (and the polyfill in this case is so small, it is not a big problem for users to copy paste it into their programs.)

I do think it would be informative to write the kernel for this, but that'll be a bit longer journey into some of the more numeric computing parts of PyTorch.
